pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -916,7 +916,9 @@ func (m *kubeGenericRuntimeManager) SyncPod(pod *v1.Pod, podStatus *kubecontaine
 	// containers cannot be specified on pod creation.
 	if utilfeature.DefaultFeatureGate.Enabled(features.EphemeralContainers) {
 		for _, idx := range podContainerChanges.EphemeralContainersToStart {
-			start("ephemeral container", metrics.EphemeralContainer, ephemeralContainerStartSpec(&pod.Spec.EphemeralContainers[idx]))
+			if err := start("ephemeral container", metrics.EphemeralContainer, ephemeralContainerStartSpec(&pod.Spec.EphemeralContainers[idx])); err != nil {
+				return
```

Comment:
See the comment above

```
// These are started "prior" to init containers to allow running ephemeral containers even when there
// are errors starting an init container. In practice init containers will start first since ephemeral
```

Reply:
This PR significantly changes pod behavior, which is:

1. Start all unstarted ephemeral containers in parallel
2. Start init containers sequentially, halting on error
3. Start all regular containers in parallel

Errors are already handled in the `start()` helper, and for ephemeral and regular containers we want to keep going in the control loop.